crates/vm/levm/src/vm.rs

```diff
@@ -410,7 +404,9 @@ impl VM {
                 calldata_cost += 4;
             }
         }
-        report.gas_used += calldata_cost;
+        
+        let max_gas = self.env.gas_limit.low_u64();
```

Comment:
`gas_limit` is `U256`, this conversion could panic. Let's operate with `U256` (this may imply updating `TransactionReport` types, which is ok).

Reply:
Will fix it in the next issue